### DISCUSSION TITLE
Video playback issue for .ts files

### DISCUSSION THREAD

**Discussion by cvscientist00** [stashapp/stash#3796] [Q&A] [answered]
Hello everyone,

I've been using the Stash App for a while now and absolutely love it. However, I'm encountering a persistent issue that I'm struggling to resolve.

I have a considerable collection of videos that I've downloaded from the internet, primarily in .mp4 and .ts formats. I'm facing difficulties with the video player for .ts files. Every time I attempt to play a .ts file, it takes several seconds for the video to start playing. Once the video starts, it runs smoothly, but if I try to skip to another point on the seek bar, I encounter the same delay before playback resumes at the new position. Interestingly, I do not experience these issues with .mp4 files, regardless of their size.

For additional context, I had previously used VLC player for video playback and encountered the same issue with .ts files, both in terms of starting and seeking. However, when I switched to the MPV player, I found that it handled both .mp4 and .ts files smoothly. This leads me to believe that my PC has adequate resources to handle .ts and .mp4 files seamlessly, suggesting that there could be some settings in the VLC player and Stash's web player that are hampering playback.

Is there a solution to this problem? Any advice or guidance would be highly appreciated. Thank you.

**Comment by Flashy78** [comment, marked as answer]
What you're describing sounds like Stash is live transcoding the file in order to play it in your web browser. Click on the little gear icon in the player, if it doesn't say `direct stream` then the file is being transcoded. Transcoding is slow to seek/skip around because it has to restart the transcoding process each time.
In the File Info tab, click the Stream link to try and play the video straight in your browser. If you browser cannot play it, then Stash must transcode it. Maybe try another browser.
VLC and MPV player do not use web browsers to play videos, so they can play videos with lots of different codecs and can ignore errors during playback much better than browsers can.

You can convert your .ts files to something more supported like .mp4 which your browser should be able to play natively.

**Reply by cvscientist00** [reply]
Thank you for detailed response. For ts files there is no "direct stream" in the player options. 
I tried the other thing you mentioned. Clicking the Stream link in File Info tab downloads a Steam.tts file in chrome/firefox and opera browser. Is there any other way to direct play .ts file in the browser from the stream link?
Maybe using some browser extension or something?

**Reply by Flashy78** [reply]
Downloading means the browser cannot play the file, so no website (including Stash) will be able to play the file. There isn't any extension for that, it's built into the browser directly. It's only recently Chrome was able to finally play HEVC files, even though Safari has been able to play them forever.
You could use something like https://docs.stashapp.cc/add-ons/userscripts/#stash-open-media-player to open the file in a media player instead of the browser, but I've never used it so not sure how well it works.
In mobile view, there is a button that shows up to open the video outside of the browser when supported.
